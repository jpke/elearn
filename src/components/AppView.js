import React from 'react';
import { Link, IndexLink } from 'react-router';

//view component, receives props from AppContainer
//returns background app view nav and header
//calls child components passed in as props
const AppView = (props) => (
  <div>
    <div className="header">
      <div>
        <IndexLink to="/">Home</IndexLink>
        {' | '}
        <Link to="/quiz">Quiz</Link>
        {' | '}
        <Link to="/lessons">Lessons</Link>
      </div>
      {props.token ? <div className="user-name"><p>Logged in as &nbsp;</p><p>{props.userName}</p></div> : ""}
    </div>
      <br/>
      <h1>Learn Online</h1>
      {props.children}
  </div>
);

export default AppView;
