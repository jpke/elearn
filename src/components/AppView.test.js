/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';

import AppView from './AppView';

it('renders correctly', () => {
  const tree = renderer.create(
    <AppView
      userName={"currentUser"}
      token={"someToken"}
      children={[]}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
