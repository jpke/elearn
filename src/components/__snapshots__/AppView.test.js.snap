exports[`test renders correctly 1`] = `
<div>
  <div
    className="header">
    <div>
      <a
        onClick={[Function]}
        style={Object {}}>
        Home
      </a>
       | 
      <a
        onClick={[Function]}
        style={Object {}}>
        Quiz
      </a>
       | 
      <a
        onClick={[Function]}
        style={Object {}}>
        Lessons
      </a>
    </div>
    <div
      className="user-name">
      <p>
        Logged in as Â 
      </p>
      <p>
        currentUser
      </p>
    </div>
  </div>
  <br />
  <h1>
    Learn Online
  </h1>
</div>
`;
